bootstrap/bootstrap.go

```diff
@@ -48,6 +48,9 @@ type Bootstrap struct {
 
 	// Whether the checkout dir was created as part of checkout
 	createdCheckoutDir bool
+
+	// Directories to clean up at end of bootstrap
+	cleanupDirs []string
```

Comment:
Wouldn't you want the gear to stick around so in the event you want to debug something, the files are still on the FS? Is there any reason we can't rely on the OS cleaning up tmp dirs?

Reply:
I dunno that the OS cleaning up tmp dirs is a thing! It's definitely not on Linux.

Reply:
Yeah, Ruby `Tempfile` instances delete themselves when they get GCed. Otherwise, your best hope is that `/tmp` might get cleaned out on boot (very likely these days, because it's probably a tmpfs... back in the day, it was something the boot script specifically arranged to do).

We _could_ arrange to keep around the working directories for the last N builds or something, but IMO that'd be a fair bit of bother for little practical utility: I'd rather be confident that someone can manually invoke to reproduce it during their investigation. (@lox along that line, do you think it'd be useful to have an option to not do the cleanup? I don't know how often people need to jump in and read the tea-leaves of a failed build. Either way, I suspect that's more of a question for a future where the directory is potentially full of interesting things; right now it seems unlikely to contain anything of note.)

Reply:
> Yeah, Ruby Tempfile instances delete themselves when they get GCed

Ah right. Yeah I remember having to `unlink` temp files in Ruby land to have them persist shutting down the process.

> Either way, I suspect that's more of a question for a future where the directory is potentially full of interesting things; right now it seems unlikely to contain anything of note.

Yeah, totes agree. Repo-less builds is new territory, so who knows if anyone will want the files to be kept around. I have like 1/10 feels on it at the moment, so I don't mind that it's new each time! And thinking about it more, hopping in and reading the FS I think is less and less of a thing in a Dockerish world.